src/aead/chacha20_poly1305.rs

```diff
@@ -61,6 +121,49 @@ fn chacha20_poly1305_open(
     in_out: &mut [u8],
     cpu_features: cpu::Features,
 ) -> Tag {
+    #[cfg(target_arch = "x86_64")]
+    {
+        if has_sse41(cpu_features) {
+            #[repr(C)]
+            #[repr(align(16))]
```

Comment:
One thing that's unclear is whether the assembly code requires the tag buffer to be 16-byte aligned or whether it really only requires the key block to be 16-byte aligned. If the tag buffer isn't 16-byte aligned, would there be any ill consequences (performance or otherwise)?

Reply:
I think the requirement can easily be avoided using unaligned instructions?

Other than that for repr(C) if struct is aligned then everything is aligned.

Reply:
> I think the requirement can easily be avoided using unaligned instructions?
> 
> Other than that for repr(C) if struct is aligned then everything is aligned.

Thanks! I've updated the other PR to force the alignment so we don't have to think about this; see https://github.com/briansmith/ring/pull/1179/commits/af3fa46f2906f7961c96092a32c0ab529aa5d8c7.